include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -884,6 +888,178 @@ namespace VectorTools
   }
 
 
+  template <int dim, typename VectorType, int spacedim, int fe_degree>
+  void project_parallel (const Mapping<dim, spacedim>   &mapping,
+                         const DoFHandler<dim,spacedim> &dof,
+                         const ConstraintMatrix         &constraints,
+                         const Quadrature<dim>          &quadrature,
+                         const std::function< typename VectorType::value_type (const typename DoFHandler<dim, spacedim>::active_cell_iterator &, const unsigned int)> func,
+                         VectorType                     &vec_result)
+  {
+    const parallel::Triangulation<dim,spacedim> *parallel_tria =
+      dynamic_cast<const parallel::Triangulation<dim,spacedim>*>(&dof.get_tria());
+    Assert (parallel_tria !=0, ExcNotImplemented());
+
+    const MPI_Comm &mpi_communicator = parallel_tria->get_communicator();
+
+    typedef typename VectorType::value_type number;
+    Assert (dof.get_fe().n_components() == 1,
+            ExcDimensionMismatch(dof.get_fe().n_components(),
+                                 1));
+    Assert (vec_result.size() == dof.n_dofs(),
+            ExcDimensionMismatch (vec_result.size(), dof.n_dofs()));
+    Assert (dof.get_fe().degree == fe_degree,
+            ExcDimensionMismatch(fe_degree, dof.get_fe().degree));
+
+    // set up mass matrix and right hand side
+    typename MatrixFree<dim,number>::AdditionalData additional_data;
```

Comment:
number -> Number same below but I am kinda confused how `number` is deduced when you are using Trilinos vector. Have you that?